pyanaconda/modules/subscription/subscription.py

```diff
@@ -664,9 +664,12 @@ def register_username_password_with_task(self):
         task = RegisterWithUsernamePasswordTask(rhsm_register_server_proxy=register_server_proxy,
                                                 username=username,
                                                 password=password)
-        # if the task succeeds, it means the system has been registered
+        # if the task succeeds, it means the system has been registered and subscribed
+        # (with enable_content option se to True, RHSM will handle SCA/auto-attach transparently for us)
         task.succeeded_signal.connect(
             lambda: self.set_registered(True))
+        task.succeeded_signal.connect(
+            lambda: self.set_subscription_attached(True))
         # set SCA state based on data returned by the registration task
         task.succeeded_signal.connect(
             lambda: self.set_simple_content_access_enabled(
```

Comment:
I wonder if 3 lambdas could be a reason to have a method instead...

Reply:
Thanks for the suggestion, but I think I would like to keep it this way and minimize the amount of changes. But looking at the RHEL 9 version, it should be cleaner there as the relevant code got simplified in the meantime. :)